pkg/api/login.go

```diff
@@ -57,18 +57,27 @@ func (hs *HTTPServer) LoginView(c *models.ReqContext) {
 		return
 	}
 
-	if !c.IsSignedIn {
-		c.HTML(200, ViewIndex, viewData)
-		return
-	}
+	// If we already are signed in (Via auth proxy for example)
+	if c.IsSignedIn {
```

Comment:
Need to verify that auth proxy is enabled. Would be nice to extract this in similar manner as `tryOAuthAutoLogin`.